sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1025,17 +938,23 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
   /**
    * Validates the options used for alter table commands after table and columns are resolved.
    */
-  private def checkAlterTableCommand(alter: AlterTableCommand): Unit = {
-    def checkColumnNotExists(fieldNames: Seq[String], struct: StructType): Unit = {
+  private def checkAlterTableColumnCommand(alter: AlterTableColumnCommand): Unit = {
+    def checkColumnNotExists(op: String, fieldNames: Seq[String], struct: StructType): Unit = {
       if (struct.findNestedField(fieldNames, includeCollections = true).isDefined) {
-        alter.failAnalysis(s"Cannot ${alter.operation} column, because ${fieldNames.quoted} " +
+        alter.failAnalysis(s"Cannot $op column, because ${fieldNames.quoted} " +
           s"already exists in ${struct.treeString}")
       }
     }
 
     alter match {
+      case AlterTableAddColumns(table: ResolvedTable, colsToAdd) =>
+        colsToAdd.foreach { colToAdd =>
+          checkColumnNotExists("add", colToAdd.name, table.schema)
```

Comment:
What if the to-be-added columns have duplicated names? e.g. `ADD COLUMNS a ..., a ...`

Reply:
For v1 command, this is checked in [AlterTableAddColumnsCommand](https://github.com/apache/spark/blob/c0d84e6cf1046b7944796038414ef21fe9c7e3b5/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L235), but for v2, this is not checked and it's up to the catalog implementation to handle it. Is this a bug? If so, I can fix this as a separate PR.

Reply:
Also, if this is a bug, will the fix be ported to 3.2?

I am not sure whether this PR will be ported to 3.2, so I need to know whether the fix should come before or after this PR.

Reply:
Yea I think it's a bug. We can probably fix it after this PR, as this PR refactors the related code.

Reply:
I'd like to merge this PR to 3.2, as this is the last PR to complete the ALTER TABLE command migration.

Reply:
Got it thanks!